pkg/tfgen/installation_docs.go

```diff
@@ -124,3 +144,192 @@ func applyEditRules(contentBytes []byte, docFile *DocFile) ([]byte, error) {
 	}
 	return contentBytes, nil
 }
+
+func translateCodeBlocks(contentStr string, g *Generator) (string, error) {
+
+	var returnContent string
+
+	examples := map[string]string{}
+	// Extract code blocks
+	codeFence := "```"
+	var codeBlocks []codeBlock
+	for i := 0; i < (len(contentStr) - len(codeFence)); i++ {
+		block, found := findCodeBlock(contentStr, i)
+		if found {
+			codeBlocks = append(codeBlocks, block)
+			i = block.end + 1
+		}
+	}
+	if len(codeBlocks) == 0 {
+		return contentStr, nil
+	}
+	startIndex := 0
+	outDir, err := os.MkdirTemp("", "configuration-example")
+	if err != nil {
+		return "", err
+	}
+	// In order to translate the examples with the currently-being-generated version of our provider, we will obtain
+	// mappings from the cli converter.
+	mappingsDir := filepath.Join(outDir, "mappings")
+	mappings := []tfbridge.ProviderInfo{
+		g.cliConverter().info,
```

Comment:
g.cliConverter() is stateful and is coupled to a specific sequence of Start/Finish converting examples in the schema, which is not what this is doing. I'm worried about reusing the same instance.

Reply:
I think this is okay, because `registry-docs` is a separate flag of `tfgen`. We wouldn't generate registry docs and resource docs in the same call.